### PR TITLE
Improve kw patch hub ux

### PR DESCRIPTION
**Pull request by JGBSouza**
resolve some of the issues mentioned at #880, except for the fourth improvement

### CODE REVIEW

File: documentation/man/kw.rst
Comment (rodrigosiqueira) [inline documentation/man/kw.rst:80]: Squash this commit with the previous one.

File: src/ui/patch_hub/patch_hub_core.sh
Comment (rodrigosiqueira) [inline src/ui/patch_hub/patch_hub_core.sh]: There is a good chance that `"$?"` is 0, which means success, and from this code, I don't think you intend to return a success code. Am I correct? If so, you probably need to return 22 here or any other error code.

File: src/ui/patch_hub/patch_hub_core.sh
Comment (rodrigosiqueira) [inline src/ui/patch_hub/patch_hub_core.sh:206]: Use single-quotes around `Error`.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Overall, it looks good; I just left some comments.
@davidbtadokoro is the maintainer of the patch-hub; in this sense, he has the final say about this PR.

File: src/ui/patch_hub/patch_hub_core.sh
Comment (davidbtadokoro) [inline src/ui/patch_hub/patch_hub_core.sh]: As this fallback message will one be used inside the `if` statement block, move these three lines to inside the `if` statemet block.

File: src/ui/patch_hub/patch_hub_core.sh
Comment (davidbtadokoro) [inline src/ui/patch_hub/patch_hub_core.sh:162]: Don't forget to declare this variable at the top of the `show_registered_mailing_lists()` function (below `local message_box`)

File: src/ui/patch_hub/patch_hub_core.sh
Comment (davidbtadokoro) [inline src/ui/patch_hub/patch_hub_core.sh]: Instead of ` then `, I think ` -> ` is a better approach.

File: src/ui/patch_hub/patch_hub_core.sh
Comment (davidbtadokoro) [inline src/ui/patch_hub/patch_hub_core.sh]: For this whole `if` statement **block**, there are three things you should do:

- **Number 1**:

> There is a good chance that "$?" is 0, which means success, and from this code, I don't think you intend to return a success code. Am I correct? If so, you probably need to return 22 here or any other error code.

Actually, because of the FSM in `patch-hub`, returning a code 22 ends execution. Also, I feel that we can interpret that the "Error" is that there are no lists registered, and that this error is handled inside this `if` statement block, so we don't need to return an error code.

That being said, you should replace `return $?` by a simple `return`, like it is done across the feature for this handled "Errors".

- **Number 2**:

You used an indentation with four spaces in comparison with the `if` statement indentation, so you added one more indentation than needed. In other words,

```
if [[ "${#registered_mailing_lists[@]}" -eq 0 ]]; then
    create_message_box 'Error' "${fallback_message}"
    ...
```

should be

```
if [[ "${#registered_mailing_lists[@]}" -eq 0 ]]; then
  create_message_box 'Error' "${fallback_message}"
  ...
```

- **Number 3**:

Move this whole `if` statement block to between

```
  # Load registered mailing lists from config file into array
  IFS=',' read -r -a registered_mailing_lists <<< "${lore_config['lists']}"
```

and

```
  message_box='Below, you can see the lore.kernel.org mailing lists that you have registered.'$'\n'
  message_box+='Select a mailing list to see the latest patchsets sent to it.'
```

as, at this point, it is already possible to determine if there are no lists registered.

File: src/ui/patch_hub/patch_hub_core.sh
Comment (davidbtadokoro) [inline src/ui/patch_hub/patch_hub_core.sh:169]: All these changes don't belong here, but in the third commit in the PR (the one where you introduced these lines). I see that you noticed the indentation problem I mentioned in the third commit, but this should be updated there using interactive rebase.

**Review by davidbtadokoro — CHANGES_REQUESTED** [review]
Hi @JBGSouza, and thanks so much for tackling this `patch-hub` related UX issues.

Left some comments, so please try to address them. Don't hesitate to ask me for any doubts about them or anything else.

**Comment by davidbtadokoro** [comment]
Hi @JGBSouza, and thanks for the dedication on updating this PR!

Everything looked legit to me, so I merged it into the unstable branch 👍

**Pull request closed without merging**
